src/apps/viewer/extensions/cellpack/model.ts

```diff
@@ -34,40 +35,54 @@ function getCellPackModelUrl(fileName: string, baseUrl: string) {
     return `${baseUrl}/results/${fileName}`
 }
 
-async function getModel(id: string, baseUrl: string) {
+async function getModel(id: string, source:IngredientSource, baseUrl: string) {
     let model: Model;
+    const mid = source.model? parseInt(source.model) : 0;
```

Comment:
Nitpicking, I think this is the model-index not the model-id which is a string name. If you want the model.id you would need to loop over the trajectory.

Reply:
I will try figure this out !